projects/appearances/protos/Parquetry.proto

```diff
@@ -1,39 +1,56 @@
 #VRML_SIM R2020a utf8
 # license: Apache License 2.0
 # license url: http://www.apache.org/licenses/LICENSE-2.0
-# A wooden material in a tesselated block pattern covered with a layer of varnish. Useful with the `Floor` PROTO.
+# A parquetry material. The type can be choose between "chequered", "dark strip", "light strip" and "mosaic". The color can be overridden using the `colorOverride` field.
 
 PROTO Parquetry [
-  field SFColor colorOverride    1 1 1  # Defines the default color multiplied with the texture color.
-  field SFNode  textureTransform NULL   # Defines an optional 2d texture transform.
-  field SFFloat IBLStrength      1      # Defines the strength of ambient lighting from the Background node.
+  field SFString{"chequered", "dark strip", "light strip", "mosaic"} type             "mosaic"  # Defines the parquetry type.
+  field SFColor                                                      colorOverride    1 1 1     # Defines the default color multiplied with the texture color.
+  field SFNode                                                       textureTransform NULL      # Defines an optional 2d texture transform.
+  field SFFloat                                                      IBLStrength      1         # Defines the strength of ambient lighting from the Background node.
 ]
 {
+%{
+  local textureName = fields.type.value:gsub("%s+", "_")
+  local extension = "jpg\""
+  if fields.type.value == "light strip" then
+    extension = "png\""
+  end
+}%
   PBRAppearance {
     baseColor IS colorOverride
     baseColorMap ImageTexture {
       url [
-        "textures/parquetry/parquetry_base_color.jpg"
+        %{= "\"textures/parquetry/" .. textureName .. "_parquetry_base_color." .. extension }%
       ]
+      filtering 5
```

Comment:
Is there a reason for adding all these filtering?

Reply:
Some yes. I check them all again and removed useless one.